File: android/src/main/java/expo/modules/alarm/AlarmSchedulerOccurrencePolicy.kt
package expo.modules.alarm

import java.util.UUID

internal object AlarmSchedulerOccurrencePolicy {
  fun newPrimaryOccurrenceId(): String = UUID.randomUUID().toString()

  fun resolutionStorageKey(occurrenceId: String, idempotencyKey: String): String =
    "resolution:$occurrenceId:$idempotencyKey"
}

File: ios/AlarmScheduler/AlarmSchedulerOccurrencePolicy.swift
import Foundation

enum AlarmSchedulerOccurrencePolicy {
  static func newPrimaryOccurrenceId() -> String {
    UUID().uuidString
  }

  static func resolutionStorageKey(occurrenceId: String, idempotencyKey: String) -> String {
    "\(occurrenceId):\(idempotencyKey)"
  }

  static func preservesPrimarySchedule(relationship: String?) -> Bool {
    relationship == "primary"
  }
}
